### PR TITLE
Cpf 4666/permission menu dao controller for table view create and detail view edit and delete

### PR DESCRIPTION
**Pull request by olhabn**

### CODE REVIEW

File: src/foam/lib/PermissionPredicate.js [hunk 1]
```diff
@@ -29,25 +29,15 @@ foam.CLASS({
       javaCode: `
         X x = (X) obj;
         AuthService auth = (AuthService) x.get("auth");
-        if ( args_ == null || args_.length <= 0 ) {
-          return true;
-        }
-
-        for ( String permission : args_ ) {
-          if ( auth.check(x, permission) )
+        if ( auth.check(x, permission_) )
```
Comment (jlhughes) [inline src/foam/lib/PermissionPredicate.js]:
```suggestion
        return auth.check(x, permission_);
```

File: src/foam/lib/PermissionPredicate.js [hunk 2]
```diff
@@ -29,25 +29,15 @@ foam.CLASS({
...
             return true;
-        }
         return false;
       `,
       code: async function() {
-        if ( ! this.args || this.args.length <= 0 )
+        var r = await this.auth.check(null, this.permission);
```
Comment (jlhughes) [inline src/foam/lib/PermissionPredicate.js]: Can this become
```suggestion
        return await this.auth.check(null, this.permission) ? true: false; 
```

**Review by jlhughes — COMMENTED** [review]
Nice.

File: src/foam/lib/PermissionPredicate.js (hunk 1 above)
Comment (olhabn) [reply src/foam/lib/PermissionPredicate.js]: Fixed

File: src/foam/lib/PermissionPredicate.js (hunk 2 above)
Comment (olhabn) [reply src/foam/lib/PermissionPredicate.js]: Fixed

**Review by jlhughes — APPROVED** [review]

**Comment by kgrgreer** [comment]
Looks good, but has has merge conflicts.  Please resolve.

**Pull request merged**